packages/portfolio-contract/src/portfolio.flows.ts

```diff
@@ -916,29 +917,26 @@ export const makeLCA = (async (orch: Orchestrator): Promise<LocalAccount> => {
 }) satisfies OrchestrationFlow;
 harden(makeLCA);
 
-export const withdraw = (async (
+export const executePlan = (async (
   orch: Orchestrator,
   ctx: PortfolioInstanceContext,
   seat: ZCFSeat,
+  _offerArgs: unknown,
   pKit: GuestInterface<PortfolioKit>,
-) => {
+  flowDetail: FlowDetail,
+): Promise<`flow${number}`> => {
   const pId = pKit.reader.getPortfolioId();
-  const traceP = makeTracer('withdraw').sub(`portfolio${pId}`);
-
-  // cast justified by proposal shape.
-  const proposal = seat.getProposal() as unknown as ProposalType['withdraw'];
-  const { Cash: amount } = proposal.want;
+  const traceP = makeTracer(flowDetail.type).sub(`portfolio${pId}`);
 
   // XXX enhancement: let caller supply steps
-  const { stepsP, flowId } = pKit.manager.startFlow({
-    type: 'withdraw',
-    amount,
-  });
+  const { stepsP, flowId } = pKit.manager.startFlow(flowDetail);
   const traceFlow = traceP.sub(`flow${flowId}`);
   traceFlow('waiting for steps from planner');
+  // idea: race with seat.getSubscriber()
   const steps = await (stepsP as unknown as Promise<MovementDesc[]>); // XXX Guest/Host types UNTIL #9822
   try {
```

Comment:
because of the finishFlow, I think everything after startFlow should be in the `try`